### PR TITLE
feat: List process registry query

### PR DESCRIPTION
**Pull request by alexrodfe**
## Description

- Added graphql query for listing process registries.
- Added usecase and repo support for listing query.
- Added fix for collections index operation in repo layer (mongodb technology).

## Motivation and Context

The second phase of process registries commands demands support for listing operations. Users will be able from kli to query through graphql process registries from a product. This query can have a process type filter.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other changes (ci configuration, documentation or any other kind of changes)

## Checklist:

- [X] I have created tests for my code changes, and the tests are passing.
- [ ] I have executed the pre-commit hooks locally.
- [ ] My change requires a change to the documentation (create a new issue if the documentation has not been updated).
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by kafkaphoenix — APPROVED** [review]

File: engine/admin-api/adapter/gql/resolver.go [hunk 1]
```diff
@@ -210,6 +210,17 @@ func (r *queryResolver) Versions(ctx context.Context, productID string) ([]*enti
 	return r.versionInteractor.ListVersionsByProduct(ctx, loggedUser, productID)
 }
 
+func (r *queryResolver) ProcessRegistries(ctx context.Context, productID string, processType *string) ([]*entity.ProcessRegistry, error) {
```
Comment (dfinteligenz) [inline engine/admin-api/adapter/gql/resolver.go:213]: We are retrieving a list of processes (ore registered processes), not process registries.
I'd rename the method to reflect that.

```suggestion
func (r *queryResolver) RegisteredProcesses(ctx context.Context, productID string, processType *string) ([]*entity.ProcessRegistry, error) {
```

File: engine/admin-api/domain/repository/process_registry.go [hunk 2]
```diff
@@ -9,6 +9,7 @@ import (
 )
 
 type ProcessRegistryRepo interface {
-	Create(productID string, newProcessRegistry *entity.ProcessRegistry) (*entity.ProcessRegistry, error)
 	CreateIndexes(ctx context.Context, productID string) error
+	Create(productID string, newProcessRegistry *entity.ProcessRegistry) (*entity.ProcessRegistry, error)
+	ListByProductWithTypeFilter(ctx context.Context, productID, processType string) ([]*entity.ProcessRegistry, error)
```
Comment (dfinteligenz) [inline engine/admin-api/domain/repository/process_registry.go:14]: Both product and type are filters for the query, I'd rename the method to `ListByProductAndType` to make it more clear.

```suggestion
	ListByProductAndType(ctx context.Context, productID, processType string) ([]*entity.ProcessRegistry, error)
```

File: engine/admin-api/domain/usecase/process_service.go [hunk 3]
```diff
@@ -98,11 +95,27 @@ func (ps *ProcessService) RegisterProcess(
 	return processID, nil
 }
 
-func (ps *ProcessService) getProcessID(processRef string) (string, error) {
+func (ps *ProcessService) getProcessID(processRef string) string {
 	split := strings.Split(processRef, "/")
+
 	if len(split) != 2 {
-		return "", ErrInvalidProcessReference
+		ps.logger.Error(ErrInvalidProcessReference, "%q, defaulting to use whole process reference", processRef)
```
Comment (dfinteligenz) [inline engine/admin-api/domain/usecase/process_service.go:102]: As the default behavior is to return a default value, I'd replace the Error log with a Warn log instead.

File: engine/admin-api/schema.graphql [hunk 4]
```diff
@@ -5,6 +5,7 @@ type Query {
   products: [Product!]!
   version(name: String!, productID: ID!): Version!
   versions(productID: ID!): [Version!]!
+  processRegistries(productID: ID!, processType: String): [ProcessRegistry]!
```
Comment (dfinteligenz) [inline engine/admin-api/schema.graphql:8]: Ditto, I'd replace the method with `registeredProcesses`.

```suggestion
  registeredProcesses(productID: ID!, processType: String): [ProcessRegistry]!
```

**Review by dfinteligenz — CHANGES_REQUESTED** [review]
Great job!
Just a pair of suggestions.

File: engine/admin-api/domain/usecase/process_service.go (hunk 3 above)
Comment (alexrodfe) [reply engine/admin-api/domain/usecase/process_service.go]: There is no logger.Warn so I'll use a Info with a "WARNING" in string

**Review by JYisus — APPROVED** [review]

File: engine/admin-api/domain/usecase/process_service.go (hunk 3 above)
Comment (dfinteligenz) [reply engine/admin-api/domain/usecase/process_service.go]: You need to add a V(2) if I'm not mistaken so that the level of the logger is WARN.

**Review by dfinteligenz — APPROVED** [review]

**Pull request merged**